app/controllers/exercises_controller.rb

```diff
@@ -24,6 +25,10 @@ def set_default_content
     @default_content = @exercise.default_content_for(current_user) if current_user?
   end
 
+  def set_comments
+    @comments = @exercise.comments_for(current_user) if current_user?
+    @comments.try(:each, &:mark_as_readed)
+  end
```

Comment:
If you do the each within the if and `tap`, you don't need to use try and an extra variable